nilearn/datasets/func.py

```diff
@@ -1559,3 +1660,523 @@ def fetch_cobre(n_subjects=10, data_dir=None, url=None, verbose=1):
+            Matched image metadata will trigger actual image download.
+        If a dict, each key-value pair is an API filter.
+            The key will be checked for equality to the value.
+            Keys are applied before metadata download, limiting
+            the number of rows returned and possibly the number
+            of round trips made.
+        (see http://neurovault.org/api-docs#collapseImages for
+         API keys and collection metadata keys)
+
+    data_dir: string, optional (default None)
+        Path of the data directory. Used to force data storage in a specified
+        location. Default: None.
+
+    url: string, optional (default None)
+        Override download URL. Used for test only (or if you setup a mirror of
+        the data).
+
+    overwrite: bool, optional (default False)
+        If True, re-download cached image metadata and data.
+
+    resume: bool, optional (default True)
+        If True, try resuming download if possible.
+
+    verbose: int, optional (default 0)
+        Defines the level of verbosity of the output.
+
+    Returns
+    -------
+    data: Bunch
+        Dictionary-like object, the interest attributes are :
+        'collections': dict of dicts (one per collection)
+            Metadata about each collection (key: collection ID)
+            See http://neurovault.org/api-docs#collapseCollections
+            for all available fields.
+        'images': list of dicts
+            Metadata of image; parallel array to func_files.
+            See http://neurovault.org/api-docs#collapseImages
+            for available fields.
+            Also includes `local_path` (path to downloaded image)
+            and `collection_id` (which indexes into `collections`)
+        'func_files': list of strings
+            Paths to betamaps.
+
+    References
+    ----------
+    [1] Gorgolewski KJ, Varoquaux G, Rivera G, Schwartz Y, Ghosh SS, Maumet C,
+        Sochat VV, Nichols TE, Poldrack RA, Poline J-B, Yarkoni T and
+        Margulies DS (2015) NeuroVault.org: a web-based repository for
+        collecting and sharing unthresholded statistical maps of the human
+        brain. Front. Neuroinform. 9:8.
+        doi: 10.3389/fninf.2015.00008
+    """
+
+    # Massage parameters, convert into image filters.
+    if exclude_known_bad_images:
+        bad_collects = [16]
+        bad_image_ids = [
+            96, 97, 98,                    # The following maps are not brain maps
+            338, 339,                      # And the following are crap
+            335,                           # 335 is a duplicate of 336
+            3360, 3362, 3364,              # These are mean images, and not Z maps
+            1202, 1163, 1931, 1101, 1099]  # Ugly / obviously not Z maps
+        collection_ids = chain(collection_ids, [-bid for bid in bad_collects])
+        image_ids = chain(image_ids, [-bid for bid in bad_image_ids])
+        image_filters = chain(image_filters,
```

Comment:
passing a dictionary as image_filters or collections_filters will not work because it is always used as if it were a tuple of callables (the other possibility for filtering) anyway. 
e.g.
`fetch_neurovault(max_images=10, collection_filters={'DOI': '10.1016/j.neurobiolaging.2012.11.002'})`